### PR TITLE
updated side-nav, user icon, search text, logo and text positioning, …

### PR DESCRIPTION
**Pull request by nhieubrian**
…and added template for fav recipes

# Pull Request Process
1. Ensure that your branch is as up-to-date with main as possible 
2. Fill out this template
3. Have at least 2 other people to get PR approval comments (could just add everybody on team)
4. Create PR

# Changes
## What changes did you make?

- Fixed side-nav to be empty red block
- Updated positioning for user icon (more centered)
- Updated search text (b/c I don't think we will be searching for favorited recipes)
- Updated logo and logo text positioning, now does not move when scrolling
- Created recipe template for favorited recipes, includes title, picture, and time.

# Verification
## How did you ensure that your changes work?

Tested on local machine. Template was tested with long title, overflow is prevented. 

TODO: Update for resized windows and use Local Storage values.

# Screenshots
## Please post screenshots of your changes working if possible

5d34bfedfe3cb483ffa2215ba27e4baa

### CODE REVIEW

File: source/user-portal/assets/favRecipe.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        const recipeTitle = document.createElement('a');
+        recipeTitle.setAttribute('href',"");
+        recipeTitle.setAttribute('class','title');
+
+        const recipeImg = document.createElement('img');
+        const timeCircle = document.createElement('div');
+        timeCircle.setAttribute('class', 'time');
+
+        const timeNumb = document.createElement('p')
+        timeNumb.setAttribute('class', 'recipe-time');
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:88]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
        timeNumb.setAttribute("class", 'recipe-time');
```

File: source/user-portal/assets/favRecipe.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        const recipe = document.createElement('article');
+
+        const recipeTitle = document.createElement('a');
+        recipeTitle.setAttribute('href',"");
+        recipeTitle.setAttribute('class','title');
+
+        const recipeImg = document.createElement('img');
+        const timeCircle = document.createElement('div');
+        timeCircle.setAttribute('class', 'time');
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:85]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
        timeCircle.setAttribute("class", 'time');
```

File: source/user-portal/assets/script.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    createRecipeCards();
+}
+
+async function fetchRecipes() {
+    return new Promise((resolve, reject) => {
+
+        for(let i = 0; i < recipes.length; i++){
+            fetch(recipes[i])
+            .then(response => response.json())
+            .then(data => {
```
Comment (nlpham) [inline source/user-portal/assets/script.js:31]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
            .then((data) => {
```

File: source/user-portal/assets/favRecipe.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
+class FavRecipe extends HTMLElement {
+    constructor() {
+      super();
+      this.attachShadow({mode: 'open'});
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
      this.attachShadow({mode: "open"});
```

File: source/user-portal/assets/favRecipe.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            }
+        `;
+        
+        styleElem.innerHTML = styles;
+
+        const recipe = document.createElement('article');
+
+        const recipeTitle = document.createElement('a');
+        recipeTitle.setAttribute('href',"");
+        recipeTitle.setAttribute('class','title');
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:81]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
        recipeTitle.setAttribute("class",'title');
```

File: source/user-portal/assets/favRecipe.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        const recipeImg = document.createElement('img');
+        const timeCircle = document.createElement('div');
+        timeCircle.setAttribute('class', 'time');
+
+        const timeNumb = document.createElement('p')
+        timeNumb.setAttribute('class', 'recipe-time');
+
+        timeNumb.innerHTML = "0";
+
+        recipeImg.setAttribute('src','images/sadburger.gif');
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:92]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
        recipeImg.setAttribute("src",'images/sadburger.gif');
```

File: source/user-portal/assets/favRecipe.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
+class FavRecipe extends HTMLElement {
+    constructor() {
+      super();
+      this.attachShadow({mode: 'open'});
+    }
+  
+    set data(data) {
+        const styleElem = document.createElement('style');
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
        const styleElem = document.createElement("style");
```

File: source/user-portal/assets/script.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                recipeData[i] = data;
+                if(i + 1 == recipes.length){
+                    resolve(true);
+                }
+            })
+            .catch((error) => {
+                console.error('Error fetching: ', error);
```
Comment (nlpham) [inline source/user-portal/assets/script.js:38]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
                console.error("Error fetching: ", error);
```

File: source/user-portal/assets/script.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+    createRecipeCards();
+}
+
+async function fetchRecipes() {
+    return new Promise((resolve, reject) => {
+
+        for(let i = 0; i < recipes.length; i++){
+            fetch(recipes[i])
+            .then(response => response.json())
```
Comment (nlpham) [inline source/user-portal/assets/script.js:30]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
            .then((response) => response.json())
```

File: source/user-portal/assets/favRecipe.js [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+            .recipe-time {
+                color: white;
+                margin-left:12px;
+                text-align: center;
+            }
+        `;
+        
+        styleElem.innerHTML = styles;
+
+        const recipe = document.createElement('article');
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:77]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
        const recipe = document.createElement("article");
```

File: source/user-portal/assets/script.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+function createRecipeCards(){
+
+    for(let i = 0; i < 6; i++){
+
+        let card = document.createElement('fav-recipe');
+        
+        card.data = "Favorite more recipes!";
+
+        document.querySelector('recipes').appendChild(card);
```
Comment (nlpham) [inline source/user-portal/assets/script.js:54]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
        document.querySelector("recipes").appendChild(card);
```

File: source/user-portal/assets/favRecipe.js [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+        const recipeTitle = document.createElement('a');
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:79]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8429335)
```suggestion
        const recipeTitle = document.createElement("a");
```

**Review by limpa105 — APPROVED** [review]
This Looks great to me! I like the fact that we have a template for the recipe cards! Good job guys!

**Review by renaissancejlc — APPROVED** [review]

**Pull request merged**